AndroidNativeLoginTemplate/app/src/main/java/com/salesforce/androidnativelogintemplate/NativeLogin.kt

```diff
@@ -161,167 +199,516 @@ class NativeLogin : ComponentActivity() {
+                                    colorFilter = ColorFilter.tint(colorScheme.primary),
+                                    contentDescription = "Back",
+                                    modifier = Modifier.padding(start = 16.dp)
+                                )
+                            }
+                        }
+                    }
+                },
+                bottomBar = {
+                    Column(
+                        horizontalAlignment = CenterHorizontally,
+                        modifier = Modifier
+                            .navigationBarsPadding()
+                            .fillMaxWidth(),
+                    ) {
+                        // Fallback to web based authentication.
+                        TextButton(onClick = { handleWebviewFallbackResult?.launch(webviewLoginIntent) }) {
+                            Text(text = "Looking for Salesforce Log In?")
+                        }
+                    }
+                },
+            ) { innerPadding ->
                 Column(
-                    horizontalAlignment = Alignment.CenterHorizontally,
+                    horizontalAlignment = CenterHorizontally,
                     modifier = Modifier
+                        .verticalScroll(rememberScrollState(), reverseScrolling = true)
                         .navigationBarsPadding()
-                        .fillMaxWidth(),
+                        .padding(top = innerPadding.calculateTopPadding())
+                        .imePadding()
+                        .fillMaxSize(),
                 ) {
-                    // Fallback to web based authentication.
-                    TextButton(onClick = { handleWebviewFallbackResult?.launch(webviewLoginIntent) }) {
-                        Text(text = "Looking for Salesforce Log In?")
+                    Spacer(modifier = Modifier.height(75.dp))
+                    ElevatedCard(
+                        modifier = Modifier
+                            .padding(16.dp)
+                            .navigationBarsPadding()
+                            .widthIn(350.dp, 500.dp),
+                        elevation = CardDefaults.cardElevation(defaultElevation = 6.dp)
+                    ) {
+                        Spacer(modifier = Modifier.height(25.dp))
+                        Image(
+                            painter = painterResource(id = sf__salesforce_logo),
+                            colorFilter = ColorFilter.tint(colorScheme.primary),
+                            contentDescription = "",
+                            modifier = Modifier.align(CenterHorizontally),
+                        )
+
+                        // Switch the layout to match the selected identity flow.
+                        when (identityFlowLayoutTypeActive) {
+                            LoginViaUsernamePassword -> UserNamePasswordInput(login)
+
+                            InitializePasswordLessLoginViaOtp -> UserNameOtpVerificationMethodInput()
```

Comment:
It's almost impossible to tell in Github's diff, but the two new "input" composable methods don't need any callbacks or state passed to them.  This is because these methods are members of the activity and can directly call its methods and view model.  As the number of parameters to things got higher, this helped avoid adding connecting parameters for every input and output.

The diff in Android Studio for this file is much easier to read, of course.